patches/tModLoader/Terraria/ModLoader/GlobalWall.cs

```diff
@@ -24,5 +24,17 @@ public abstract class GlobalWall : GlobalBlockType
 		/// </summary>
 		public virtual void KillWall(int i, int j, int type, ref bool fail) {
 		}
+
+		/// <summary>
+		/// Called for every wall that updates due to being placed or being next to a wall that is changed. Return false to stop the game from carrying out its default WallFrame operations. Returns true by default.
+		/// </summary>
+		/// <param name="i">The x position in tile coordinates.</param>
+		/// <param name="j">The y position in tile coordinates.</param>
+		/// <param name="type">Type of the wall being framed</param>
+		/// <param name="resetFrame">Whether this wall should try to change it's display frame</param>
+		/// <returns></returns>
+		public virtual bool WallFrame(int i, int j, int type, ref bool resetFrame) {
```

Comment:
I don't think `resetFrame` should be `ref`

The documentation for `resetFrame` should say "randomize frame" since that's what it's actually for.